react-file-uploads/client/src/UploadForm.tsx

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { type FormEvent, useState } from 'react';
+
+type Image = {
+  imageId: number;
+  url: string;
+  caption: string;
+};
+
+export function UploadForm() {
+  async function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    /* Prevent the browser's default behavior for form submissions.
+     * Create a `new` FormData object from the `event`.
+     *
+     * Use fetch() to send a POST request to /api/uploads.
+     * The body should be the form data object you created (not a JSON string).
+     * Headers are not necessary as fetch will use the correct Content-Type
+     * automatically (multipart/form-data).
+     *
+     * Parse the JSON response body and log the parsed response body.
+     * Remember to use type assertion when retrieving the JSON from the response.
+     * Set the `imageFile` state to the parsed response body.
+     * Log any errors to the console (using `console.error`).
+     *
+     * References:
+     * https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData
+     * https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_a_file
+     */
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
```

Comment:
This function should be wrapped in try/catch so you can catch and report errors. If an error occurs (in the catch block), you can just use alert() to display the error.